python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -164,74 +172,100 @@ def run(self, templateExposure, exposure, subtractedExposure, psfMatchingKernel)
                 spatialKernel = psfMatchingKernel
                 kimg = afwImage.ImageD(spatialKernel.getDimensions())
                 bbox = subtractedExposure.getBBox()
-                xcen = (bbox.getBeginX() + bbox.getEndX()) / 2.
-                ycen = (bbox.getBeginY() + bbox.getEndY()) / 2.
+                if xcen is None:
+                    xcen = (bbox.getBeginX() + bbox.getEndX()) / 2.
+                if ycen is None:
+                    ycen = (bbox.getBeginY() + bbox.getEndY()) / 2.
+                self.log.info("Using matching kernel computed at (%d, %d)" % (xcen, ycen))
                 spatialKernel.computeImage(kimg, True, xcen, ycen)
-        except:  # Not a spatially-varying kernel (usually not true)
-            kimg = psfMatchingKernel.computeImage()
 
-        # Compute the images' sigmas (sqrt of variance)
-        statObj = afwMath.makeStatistics(templateExposure.getMaskedImage().getVariance(), afwMath.MEANCLIP,
-                                         self.statsControl)
-        sig1 = np.sqrt(statObj.getValue(afwMath.MEANCLIP))
+                if False:  # debug code to save spatially varying kernel for analysis
+                    import cPickle
+                    import gzip
+                    cPickle.dump(spatialKernel, gzip.GzipFile('spatialKernel.p.gz', 'wb'))
 
-        statObj = afwMath.makeStatistics(exposure.getMaskedImage().getVariance(), afwMath.MEANCLIP,
-                                         self.statsControl)
-        sig2 = np.sqrt(statObj.getValue(afwMath.MEANCLIP))
+        except:  # Not a spatially-varying kernel (usually not true)
+            kimg = psfMatchingKernel.computeImage()
 
-        corrKernel = DecorrelateALKernelTask._computeDecorrelationKernel(kimg.getArray(), sig1=sig1, sig2=sig2)
-        fcorrKernel = DecorrelateALKernelTask._fixEvenKernel(corrKernel)
+        if var1 is None:
+            statObj = afwMath.makeStatistics(exposure.getMaskedImage().getVariance(),
```

Comment:
You could make a short helper function that takes an Exposure and a statsControl and just returns the variance and call it in these two `if None` blocks, since they do the same thing (and I'm guessing you want them to always do the same thing).